torch/backends/_nnapi/serializer.py

```diff
@@ -1665,10 +1685,55 @@ def add_conv2d_common(
             inputs[10] = self.add_immediate_bool_scalar(use_nchw)
 
         outputs = [None] * 1
-        outputs[0] = self.add_tensor_operand(jit_out, out_oper)
+        out_shape = get_conv_pool_shape(image_oper.shape, args, out_c, transpose)
+        out_oper = image_oper._replace(
+            shape=out_shape,
+            scale=out_scale,
+            zero_point=out_zero_point,
+        )
+        out_id = self.add_tensor_operand(jit_out, out_oper)
+        self._handle_conv_pool_flexible_input(out_id, jit_image, args, transpose)
 
+        outputs[0] = out_id
         self.add_operation(opcode, inputs, outputs)
 
+    def _handle_conv_pool_flexible_input(self, out_id, jit_image, args, transpose):
```

Comment:
I wonder if `get_conv_pool_shape` should be moved near this since they're tightly coupled.